source/views/menus/menu-bonapp.tsx

```diff
@@ -150,93 +150,108 @@ function prepareFood(cafeMenu: MenuInfoType) {
+		headers: {accept: 'application/json'},
+	})
+}
+
+function getCafeThing(cafe: Props['cafe']) {
+	return typeof cafe === 'string' ? cafe : cafe.id
+}
+
+function buildUrls(cafeThing: Props['cafe']) {
+	let cafe = getCafeThing(cafeThing)
+
+	let cafeUrl = undefined
+	let menuUrl = undefined
+
+	if (typeof cafeThing === 'string') {
+		cafeUrl = API(`/food/named/cafe/${cafe}`)
+		menuUrl = API(`/food/named/menu/${cafe}`)
+	} else if ('id' in cafeThing) {
+		cafeUrl = API(`/food/cafe/${cafe}`)
+		menuUrl = API(`/food/menu/${cafe}`)
+	} else {
+		throw new Error('invalid cafe passed to BonappMenu!')
 	}
 
-	let fetchData = async (cafeThing: Props['cafe'], reload?: boolean) => {
-		let menuUrl
-		let cafeUrl
-		let cafe = typeof cafeThing === 'string' ? cafeThing : cafeThing.id
-		if (typeof cafeThing === 'string') {
-			menuUrl = API(`/food/named/menu/${cafe}`)
-			cafeUrl = API(`/food/named/cafe/${cafe}`)
-		} else if ('id' in cafeThing) {
-			menuUrl = API(`/food/menu/${cafe}`)
-			cafeUrl = API(`/food/cafe/${cafe}`)
-		} else {
-			throw new Error('invalid cafe passed to BonappMenu!')
-		}
+	return {cafeUrl, menuUrl}
+}
 
-		try {
-			let delay = reload ? 500 : 0
-
-			let cafeMenuPromise = fetch(menuUrl, {delay}).json<MenuInfoType>()
-			let cafeInfoPromise = fetch(cafeUrl, {delay}).json<CafeInfoType>()
-
-			let [cafeMenu, cafeInfo] = await Promise.all([
-				cafeMenuPromise,
-				cafeInfoPromise,
-			])
-
-			setCafeMenu(cafeMenu ?? null)
-			setCafeInfo(cafeInfo ?? null)
-			setNow(moment.tz(timezone()))
-		} catch (error) {
-			if (!(error instanceof Error)) {
-				setErrorMessage('Unknown Error: Not an Error')
-				return
-			}
-			if (error.message === "JSON Parse error: Unrecognized token '<'") {
-				setErrorMessage(BONAPP_HTML_ERROR_CODE)
-			} else {
-				setErrorMessage(error.message)
-			}
-		}
+function getErrorMessage(error: Error | undefined) {
+	if (!(error instanceof Error)) {
+		return 'Unknown Error: Not an Error'
 	}
 
-	let refresh = async (): Promise<void> => {
-		setRefreshing(true)
-		await fetchData(props.cafe, true)
-		setRefreshing(false)
+	if (error.message === "JSON Parse error: Unrecognized token '<'") {
+		return BONAPP_HTML_ERROR_CODE
+	} else {
+		return error.message
 	}
+}
+
+export function BonAppHostedMenu(props: Props): JSX.Element {
+	let now = moment.tz(timezone())
+	let {cafeUrl, menuUrl} = buildUrls(props.cafe)
+	let [errorMessage, setErrorMessage] = useState<string | null>(null)
 
-	if (loading) {
+	let {
+		data: cafeMenu,
+		error: menuError,
+		reload: menuReload,
+		isPending: isMenuPending,
+		isInitial: isMenuInitial,
+		isLoading: isMenuLoading,
+	} = useCafeMenu(menuUrl)
+
+	let {
+		data: cafeInfo,
+		error: cafeError,
+		reload: cafeReload,
+		isPending: isCafePending,
+		isInitial: isCafeInitial,
+		isLoading: isCafeLoading,
+	} = useCafeInfo(cafeUrl)
+
+	useEffect(() => {
+		;[cafeError, menuError]
+			.filter((error) => error)
+			.map(getErrorMessage)
+			.forEach(setErrorMessage)
```

Comment:
I feel like this would flow better as two if-statements…? `if (cafeError) { setErrorMessage(getErrorMessage(cafeError)) }` and `if (menuError) { setErrorMessage(getErrorMessage(menuError)) }`

Reply:
Nice, that is simpler. Let's go with that!